src/routes/TradingLeague/TradingLeague.tsx

```diff
@@ -1,17 +1,79 @@
-import React, { useEffect } from 'react';
-
+import React, { useEffect, useState } from 'react';
 import Box from '@mui/material/Box';
-
 import { setPageTitle } from '@utilities';
-
 import { Page } from '@components/interface';
-import ConnectedRankingTable from '../../components/containers/ConnectedRankingTable/ConnectedRankingTable';
-import { useWallet } from '@hooks';
+import { useCurrentSeason, useWallet } from '@hooks';
+import { RankType } from '../../utilities/data';
+import { getActivity } from '@graphql';
+import Leaderboard from './Leaderboard/Leaderboard';
+const PER_PAGE = 10;
 
 const TradingLeague: React.FunctionComponent = () => {
   const { account } = useWallet();
+  const [loading, setLoading] = useState(true);
+  const season = useCurrentSeason();
+  const [rankingResults, setRankingResults] = useState<RankType[]>([]);
+  const wallet = useWallet();
+  const [userRank, setUserRank] = useState<number>(-1);
+  const [userPoints, setUserPoints] = useState<number>(-1);
+  const [page, setPage] = useState<number>(0);
+  const [ranking, setRanking] = useState<Map<string, number>>();
+  const maxPages = Math.floor(rankingResults.length / PER_PAGE) + 1;
+
+  const handleOnNextPage = () => {
+    if (page + 1 < maxPages) {
+      setPage(page + 1);
+    }
+  };
+
+  const handleOnPrevPage = () => {
+    if (page - 1 > -1) {
+      setPage(page - 1);
+    }
+  };
+
+  const fetchRankings = async () => {
```

Comment:
@ssbarbee might be also worth caching the rankings as well so that when i navigate back and forth from the league page, there is no need to refetch

Reply:
though need to be mindful of the fact that the points and rank can change over time, maybe this is a stretch but could we use websockets in this context?

Reply:
as you mentioned this is why caching won't work nicely here. Ideally yes websockets are nice to be used here but do we have the infra to do so? For now I think it is quite good, lets add the "live" feature if we have a "live" tournament requirment of sort :D I don't think anyone wants to sit in the leaderboard now and see how they climb

Reply:
yeah, websockets are a stretch at this stage, in theory also could have a button in the ui that let's users manually trigger cache refresh so it doesn't automatically refresh after going back and forth from one screen to leaderboard, though again keen to discuss other ideas around how the UX should work with Brokoli's involvement as well?